core/src/ser.rs

```diff
@@ -223,14 +239,31 @@ pub trait ByteFormat {
         W: Write;
 
     /// Read a sequence of exactly `limit` objects from the reader.
-    fn read_seq_from<R>(reader: &mut R, limit: usize) -> Result<Vec<Self>, Self::Error>
+    fn read_seq_from<R>(reader: &mut R, mode: ReadSequenceMode) -> Result<Vec<Self>, Self::Error>
     where
         R: Read,
         Self: std::marker::Sized,
     {
         let mut v = vec![];
-        for _ in 0..limit {
-            v.push(Self::read_from(reader)?);
+        match mode {
+            ReadSequenceMode::Exactly(number) => {
+                for _ in 0..number {
+                    v.push(Self::read_from(reader)?);
+                }
+                if v.len() != number {
+                    return Err(SerError::InsufficientSeqItems {got: v.len(), expected: number }.into())
```

Comment:
Did you run this through `rustfmt`? There is unequal spacing inside of the curly braces, its not padded on the opening and padded on the closing.

Reply:
have now